packages/runtime/src/runtimes/react/index.tsx

```diff
@@ -85,17 +95,10 @@ export function RuntimeProvider({
       preloadedState: contextDefaultValue.getState(),
       rootElements,
     })
-    const runtime = createReactRuntime(store)
-
-    registerBuiltinComponents(runtime)
 
     return store
   })
 
-  useEffect(() => {
```

Comment:
Won't removing this effect cause issues where the store is out of sync? We create a new store when we connect to the builder and if we don't register builtin components to that new store, how are builtin components rendered?

Reply:
So the way it's working is that when we initialize a new store, we get do `getState()` of the previous store, and use it in the `preloadedState`:
```tsx
  const [store, setStore] = useState(() => {
    const store = ReactPage.configureStore({
      preloadedState: contextDefaultValue.getState(),
      rootElements,
    })

    return store
  })
```

Reply:
Ah, that's right. We needed to get rid of the effect anyway because we want component registration to work statically so we can do things like introspection on an API endpoint (like we are now) and server components